index.js

```diff
@@ -1,500 +1,589 @@
-      console.log('errData', err, newIssueData);
-      return callback(err);
-    }
-  });
-}
 
+  //
+  // Issue Labels
+  //
 
-function makeCorrectState(ghIssueData, item, callback) {
-  if (item.state != 'closed' || ghIssueData.state == 'closed') {
-    // standard is open so we don't have to update
-    return callback(null, ghIssueData);
+  // make sure to add any labels that existed in GitLab
+  if (issue.labels) {
+    props.labels = issue.labels;
   }
 
-  // TODO get props
-  var props = {
-    owner: settings.github.owner,
-    repo: settings.github.repo,
-    number: ghIssueData.number,
-    state: 'closed',
-  };
-  if (item.milestone) {
-    var title = findMileStoneforTitle(milestoneData, item.milestone.title);
-    if (title !== null) {
-      props.milestone = title;
-    }
+  //
+  // Issue Attachments
+  //
+
+  // if the issue contains a url that contains "/uploads/", it is likely to
+  // have an attachment. Therefore, add the "has attachment" label.
+  if (props.body && props.body.indexOf('/uploads/') > -1) {
+    props.labels.push('has attachment');
   }
 
-  console.log('makeCorrectState', ghIssueData.number, item.state, props.milestone);
-  console.log('makeCorrectState props', props);
-  github.issues.edit(props, callback);
+  // create the GitHub issue from the GitLab issue
+  return github.issues.create(props);
 }
 
-function createAllIssueComments(projectID, issueID, newIssueData, callback) {
-  if (issueID == null) {
-    return callback();
+// ----------------------------------------------------------------------------
+
+/**
+ *
+ */
+async function createIssueComments(ghIssue, issue) {
+  // retrieve any notes/comments associated with this issue
+  try {
+    let notes = await gitlab.IssueNotes.all(settings.gitlab.projectId, issue.iid);
+
+    // if there are no notes, then there is nothing to do!
+    if (notes.length == 0) return;
+
+    // sort notes in ascending order of when they were created (by id)
+    notes = notes.sort((a, b) => a.id - b.id);
+
+    for (let note of notes) {
+
+      if ((/Status changed to .*/.test(note.body) && !/Status changed to closed by commit.*/.test(note.body)) ||
+          /changed milestone to .*/.test(note.body) ||
+          /Milestone changed to .*/.test(note.body) ||
+          /Reassigned to /.test(note.body) ||
+          /added .* labels/.test(note.body) ||
+          /Added ~.* label/.test(note.body) ||
+          /mentioned in issue.*/.test(note.body)) {
+        // Don't transfer when the state changed (this is a note in GitLab)
+      } else {
+
+        let bodyConverted = convertIssuesAndComments(note.body, note);
+
+        // process asynchronous code in sequence -- treats kind of like blocking
+        await github.issues.createComment({
+                    owner: settings.github.owner,
+                    repo: settings.github.repo,
+                    number: ghIssue.number,
+                    body: bodyConverted
+                  }).catch(x=>{});
```

Comment:
I know the original did not care about it but isn't it better at least console.log or maybe even exit program?

Reply:
good point :-)